modules/rfcParser.jsm.js

```diff
@@ -0,0 +1,276 @@
+	/** @override */
+	RfcParserI.atext = `(?:${RfcParserStd.atext}|${RfcParserI.UTF8_non_ascii})`;
+	/** @override */
+	RfcParserI.qtext = `(?:${RfcParserStd.qtext}|${RfcParserI.UTF8_non_ascii})`;
+
+	/** @readonly */
+	const TAG_PARSE_ERROR = {
+		/** @readonly */
+		ILL_FORMED: -1,
+		/** @readonly */
+		DUPLICATE: -2,
+	};
+
+	/**
+	* Parses a Tag=Value list.
+	* Specified in Section 3.2 of RFC 6376.
+	*
+	* @param {string} str
+	* @returns {Map<string, string>|number} Map of the parsed list or:
+	* - -1 if a tag-spec is ill-formed.
+	* - -2 duplicate tag names.
+	*/
+	function parseTagValueList(str) {
+		const tval = "[!-:<-~]+";
+		const tagName = "[A-Za-z][A-Za-z0-9_]*";
+		const tagValue = `(?:${tval}(?:(${RfcParserStd.WSP}|${RfcParserStd.FWS})+${tval})*)?`;
+
+		// delete optional semicolon at end
+		let listStr = str;
+		if (listStr.charAt(listStr.length - 1) === ";") {
+			listStr = listStr.substr(0, listStr.length - 1);
+		}
+
+		const array = listStr.split(";");
+		/** @type {Map<string, string>} */
+		const map = new Map();
+		for (const elem of array) {
+			// get tag name and value
+			const tmp = elem.match(new RegExp(
+				`^${RfcParserStd.FWS}?(${tagName})${RfcParserStd.FWS}?=${RfcParserStd.FWS}?(${tagValue})${RfcParserStd.FWS}?$`
+			));
+			if (tmp === null || !tmp[1] || tmp[2] === undefined) {
+				return TAG_PARSE_ERROR.ILL_FORMED;
+			}
+			const name = tmp[1];
+			const value = tmp[2];
+
+			// check that tag is no duplicate
+			if (map.has(name)) {
+				return TAG_PARSE_ERROR.DUPLICATE;
+			}
+
+			// store Tag=Value pair
+			map.set(name, value);
+		}
+
+		return map;
+	}
+
+	/**
+	* Parse a tag value stored in a Map.
+	*
+	* @param {ReadonlyMap<string, string>} map
+	* @param {string} tagName - name of the tag
+	* @param {string} patternTagValue - Pattern for the tag-value
+	* @param {number} [expType] - Type of exception to throw. 1 for DKIM header, 2 for DKIM key, 3 for general.
+	* @returns {[string, ...string[]]|null} The match from the RegExp if tag_name exists, otherwise null
+	* @throws {DKIM_SigError|DKIM_Error} Throws if tag_value does not match.
+	*/
+	function parseTagValue(map, tagName, patternTagValue, expType = 1) {
+		const tagValue = map.get(tagName);
+		// return null if tag_name doesn't exists
+		if (tagValue === undefined) {
+			return null;
+		}
+
+		const res = tagValue.match(new RegExp(`^${patternTagValue}$`));
+
+		// throw DKIM_SigError if tag_value is ill-formed
+		if (res === null) {
+			if (expType === 1) {
+				throw new DKIM_SigError(`DKIM_SIGERROR_ILLFORMED_${tagName.toUpperCase()}`);
+			} else if (expType === 2) {
+				throw new DKIM_SigError(`DKIM_SIGERROR_KEY_ILLFORMED_${tagName.toUpperCase()}`);
+			} else {
+				throw new DKIM_Error(`illformed tag ${tagName}`);
+			}
+		}
+
+		// @ts-expect-error
+		return res;
+	}
+
+	// @ts-ignore
+	const PREF_BRANCH = "extensions.dkim_verifier.";
+
+	// @ts-ignore
+	const prefs = Services.prefs.getBranch(PREF_BRANCH);
```

Comment:
Did you test if this `@ts-ignore` are actually needed?

I would recommend to use `@ts-ignore` only for ignore falsely reported problems with the global scope, as there is can frequently change if Typescript will report them.

In all other cases `@ts-expect-error` is preferred, as this will turn in an error if Typescript no longer sees an error.

Reply:
I didn't test it, I copied this from some other file.
I've removed the lines.